[E2E/Acceptance CI Failure]: Acceptance (practice-question-reviewer/review-submitted-questions)

### I have read the procedure

- [ ] I reproduced the flake on the develop branch.
- [ ] I observed this flake directly on the develop branch.
- [ ] The changes in my PR are unrelated to the failure.

### CI Test Type

Acceptance

### Which CI step failed?

Run Desktop Acceptance Test practice-question-reviewer/review-submitted-questions

### Did the test fail after an automatic rerun?

Test failed once and then passed after an automatic rerun

### Did the test fail when running on desktop, mobile, or both?

Desktop

### Stacktrace

```shell
FAIL core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts (518.033 s)
  Practice Question Reviewer
    ✕ should be able to navigate between questions to review
    ✕ should be able to review the submitted questions

  ● Practice Question Reviewer › should be able to navigate between questions to review

    Tab Submit Question not found.

      470 |
      471 |     if (!tabElement) {
    > 472 |       throw new Error(`Tab ${tabName} not found.`);
          |             ^
      473 |     }
      474 |
      475 |     // Click on the tab.

      at BaseUser.switchToTabInContributionDashboard (utilities/user/contributor.ts:472:13)
          at runMicrotasks (<anonymous>)
      at specs/practice-question-reviewer/review-submitted-questions.spec.ts:118:5

  ● Practice Question Reviewer › should be able to review the submitted questions

    Tab Submit Question not found.

      470 |
      471 |     if (!tabElement) {
    > 472 |       throw new Error(`Tab ${tabName} not found.`);
          |             ^
      473 |     }
      474 |
      475 |     // Click on the tab.

      at BaseUser.switchToTabInContributionDashboard (utilities/user/contributor.ts:472:13)
          at runMicrotasks (<anonymous>)
      at specs/practice-question-reviewer/review-submitted-questions.spec.ts:118:5

Test Suites: 1 failed, 1 total
Tests:       2 failed, 2 total
Snapshots:   0 total
Time:        519.276 s
Ran all test suites matching /core\/tests\/puppeteer-acceptance-tests\/specs\/practice-question-reviewer\/review-submitted-questions.spec.ts/i.
INFO     2026-04-15 15:27:37,349 shutdown.py:48] Shutting down.
Error: 4-15 15:27:37 +0000] [3251] [ERROR] Worker (pid:3253) was sent SIGTERM!
Error: 4-15 15:27:37 +0000] [2697] [ERROR] Worker (pid:2699) was sent SIGTERM!
Error: 4-15 15:27:37 +0000] [2866] [ERROR] Worker (pid:2878) was sent SIGTERM!
Error: 4-15 15:27:37 +0000] [3229] [ERROR] Worker (pid:3233) was sent SIGTERM!
INFO     2026-04-15 15:27:37,990 stub_util.py:321] Applying all pending transactions and saving the datastore
INFO     2026-04-15 15:27:37,990 stub_util.py:324] Saving search indexes
```

### Screenshots and/or Screen Recordings

https://github.com/user-attachments/assets/989316c5-8db0-4c58-983c-a0471f69bc11
https://github.com/user-attachments/assets/05975d4a-0570-4a8a-91fa-53722f065253
https://github.com/user-attachments/assets/1c5a4074-025d-4abd-b4a4-2fdf12d17ea4
https://github.com/user-attachments/assets/dc7616ef-f8ea-43f5-b2db-5f8b471058a3
https://github.com/user-attachments/assets/cdaa6b19-64ee-4f78-96dd-c420f33af811

### Occurrences

https://github.com/oppia/oppia/actions/runs/24451799862/job/71455766148

### Additional Information

_No response_

### Debugging document link

_No response_

--- a/core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts
+++ b/core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts
@@ -455,7 +455,15 @@ export class Contributor extends ExplorationEditor {
   async switchToTabInContributionDashboard(
     tabName: 'Translate Text' | 'My Contributions' | 'Submit Question'
   ): Promise<void> {
-    await this.page.waitForSelector(contributionTabSelector);
+    await this.page.waitForFunction(
+      (selector: string, name: string) => {
+        const tabs = Array.from(document.querySelectorAll(selector));
+        return tabs.some(tab => tab.textContent?.trim() === name);
+      },
+      {},
+      contributionTabSelector,
+      tabName
+    );
 
     // Get required tab element.
     const tabElements = await this.page.$$(contributionTabSelector);

